optuna/study.py

```diff
@@ -566,6 +567,30 @@ def _append_trial(
 
         self._storage.create_new_trial(self._study_id, template_trial=trial)
 
+    def _reseed_and_optimize_sequential(
+            self,
+            func,  # type: ObjectiveFuncType
+            n_trials,  # type: Optional[int]
+            timeout,  # type: Optional[float]
+            catch,  # type: Union[Tuple[()], Tuple[Type[Exception]]]
+            callbacks,  # type: Optional[List[Callable[[Study, structs.FrozenTrial], None]]]
+            gc_after_trial,  # type: bool
+            time_start  # type: Optional[datetime.datetime]
+    ):
+        # type: (...) -> None
+
+        _rng = getattr(self.sampler, "_rng", None)
```

Comment:
I think that this code cannot handle nested samplers (e.g., RNG field of `RandomSampler` owned by `TPESampler`).
How about adding an abstract method to `BaseSampler` such as `BaseSampler.reseed_rng` for this purpose?